src/Components/Error-Boundary/ErrorBoundary.tsx

```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import './Error-Boundary.scss';
+
+const ErrorBoundary: React.FC<{}> = ({ children }) => {
+    // fallback UI
+    const Oops = () => (
+        <h2 className="error-msg">Oops, something went wrong...</h2>
+    );
+
+    const hasError = false;
+
+    return <>{hasError ? <Oops /> : children}</>
```

Comment:
Looks like it will always return children. Components doesn't work as expected. Here is an [example](https://www.tek-tools.com/cloud/react-logging-and-error-handling-best-practices). Or you need to use try..catch here

Reply:
Yes, but current lesson was about functional components andthis question was discussed in next one, so in next HT it is rewitten as class component